Framework/Core/Windows/Devices/Spi/SpiDevice.cs

```diff
@@ -0,0 +1,285 @@
+using System;
+using System.Runtime.CompilerServices;
+
+namespace Windows.Devices.Spi
+{
+    public sealed class SpiDevice : IDisposable
+    {
+        private static string s_SpiPrefix = "SPI";
+        private static readonly SpiBusInfo s_busInfo = new SpiBusInfo();
+
+        private readonly string m_deviceId;
+        private readonly SpiConnectionSettings m_settings;
+
+        private object m_syncLock = new object();
+        private bool m_disposed = false;
+        private int m_mskPin = -1;
+        private int m_misoPin = -1;
+        private int m_mosiPin = -1;
+        private int m_spiBus = -1;
+
+        /// <summary>
+        /// Initializes a new instance of SpiDevice.
+        /// </summary>
+        /// <param name="deviceId">The unique name of the device.</param>
+        /// <param name="settings">Settings to open the device with.</param>
+        internal SpiDevice(string deviceId, SpiConnectionSettings settings)
+        {
+            // Device ID must match the index in device information.
+            // We don't have many buses, so just hard-code the valid ones instead of parsing.
+
+            var spiBusNames = GetValidBusNames();
+            for (int i = 0; i < spiBusNames.Length; ++i)
+            {
+                if (spiBusNames[i] == deviceId)
+                {
+                    m_spiBus = i;
+                }
+            }
+
+            // If we didn't find the exact device name in our pre-built bus list, bail out.
+            if (m_spiBus == -1)
+            {
+                throw new ArgumentException();
+            }
+
+            m_deviceId = deviceId.Substring(0);
+            m_settings = new SpiConnectionSettings(settings);
+
+            // Silence warnings about unused variables; these are used in interop.
+            if ((m_mskPin == 0) || (m_misoPin == 0) || (m_mosiPin == 0)) { }
+
+            InitNative();
+        }
+
+        ~SpiDevice()
+        {
+            Dispose(false);
+        }
+
+        /// <summary>
+        /// Gets the unique ID associated with the device.
+        /// </summary>
+        /// <value>The ID.</value>
+        public string DeviceId
+        {
+            get
+            {
+                lock (m_syncLock)
+                {
+                    if (m_disposed)
+                    {
+                        throw new ObjectDisposedException();
+                    }
+
+                    return m_deviceId.Substring(0);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets the connection settings for the device.
+        /// </summary>
+        /// <value>The connection settings.</value>
+        public SpiConnectionSettings ConnectionSettings
+        {
+            get
+            {
+                lock (m_syncLock)

```

Comment:
lock can be removed here as well, because if another thread is disposing this object, creating the settigns or not will result in a ObjectDisposedException anyway when the settings are used...
the only thing you are loosing is the causaility order (what happened first) which does not look that interesting...